packages/subgraph-service/contracts/utilities/AllocationManager.sol

```diff
@@ -475,4 +475,14 @@ abstract contract AllocationManager is EIP712Upgradeable, GraphDirectory, Alloca
     function _encodeAllocationProof(address _indexer, address _allocationId) internal view returns (bytes32) {
         return _hashTypedDataV4(keccak256(abi.encode(EIP712_ALLOCATION_PROOF_TYPEHASH, _indexer, _allocationId)));
     }
+
+    /**
+     * @notice Checks if an allocation is over-allocated
+     * @param _indexer The address of the indexer
+     * @param _delegationRatio The delegation ratio to consider when locking tokens
+     * @return True if the allocation is over-allocated, false otherwise
+     */
+    function _isAllocationOverAllocated(address _indexer, uint32 _delegationRatio) internal view returns (bool) {
+        return !allocationProvisionTracker.check(_graphStaking(), _indexer, _delegationRatio);
+    }
```

Comment:
```suggestion
    function _isOverAllocated(address _indexer, uint32 _delegationRatio) internal view returns (bool) {
        return !allocationProvisionTracker.check(_graphStaking(), _indexer, _delegationRatio);
    }
```

I would rename this function, it's the indexer that is over allocated not the allocation itself. 
Also I think it would be useful to expose this with an external function `function isOverAllocated(...) external ...` on the `SubgraphService` contract.

Reply:
Good idea, should we also include a `function isStaleAllocation()`?

Reply:
yes good catch!